contribs/gmf/src/directives/drawfeature.js

```diff
@@ -0,0 +1,399 @@
+       */
+      function(event) {
+        // In the case of azimut measure interaction, the feature's geometry is
+        // actually a collection (line + circle)
+        // For our purpose here, we only need the circle.
+        var geometry = /** @type {ol.geom.GeometryCollection} */
+            (event.feature.getGeometry());
+        event.feature = new ol.Feature(geometry.getGeometries()[1]);
+        this.handleDrawEnd_(ngeo.GeometryType.CIRCLE, event);
+      }, this);
+  ol.events.listen(measureAzimut,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // === RECTANGLE ===
+
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.drawRectangle = new ol.interaction.Draw({
+    type: ol.geom.GeometryType.LINE_STRING,
+    geometryFunction: function(coordinates, geometry) {
+      if (!geometry) {
+        geometry = new ol.geom.Polygon(null);
+      }
+      var start = coordinates[0];
+      var end = coordinates[1];
+      geometry.setCoordinates([
+        [start, [start[0], end[1]], end, [end[0], start[1]], start]
+      ]);
+      return geometry;
+    },
+    maxPoints: 2
+  });
+
+  var drawRectangle = this.drawRectangle;
+  this.interactions_.push(drawRectangle);
+  drawRectangle.setActive(false);
+  ngeoDecorateInteraction(drawRectangle);
+  this.map.addInteraction(drawRectangle);
+  ol.events.listen(drawRectangle, ol.interaction.DrawEventType.DRAWEND,
+      this.handleDrawEnd_.bind(this, geomType.RECTANGLE), this);
+  ol.events.listen(drawRectangle,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // === TEXT ===
+
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.drawText = new ol.interaction.Draw({
+    type: ol.geom.GeometryType.POINT
+  });
+
+  var drawText = this.drawText;
+  this.interactions_.push(drawText);
+  drawText.setActive(false);
+  ngeoDecorateInteraction(drawText);
+  this.map.addInteraction(drawText);
+  ol.events.listen(drawText, ol.interaction.DrawEventType.DRAWEND,
+      this.handleDrawEnd_.bind(this, geomType.TEXT), this);
+  ol.events.listen(drawText,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // Watch the "active" property, and disable the draw interactions
+  // when "active" gets set to false.
+  $scope.$watch(
+    function() {
+      return this.active;
+    }.bind(this),
+    function(newVal) {
+      if (newVal === false) {
+        this.interactions_.forEach(function(interaction) {
+          interaction.setActive(false);
+        }, this);
+      }
+    }.bind(this)
+  );
+
+};
+
+
+/**
+ * @param {ol.ObjectEvent} event Event.
+ * @private
+ */
+gmf.DrawfeatureController.prototype.handleActiveChange_ = function(event) {
+  var active;
+  this.interactions_.some(function(interaction) {
```

Comment:
use the return value of the `some` function instead of a new `var active`.
Something like
```js
this.active = this.interactions_.some(function(interaction) {
  return interaction.getActive();
});
```